pkg/server/core-db/types/Relationship.go

```diff
@@ -0,0 +1,19 @@
+package types
+
+type Cardinality string
+
+const (
+	OneToOne   Cardinality = "OneToOne"
+	OneToMany  Cardinality = "OneToMany"
+	ManyToOne  Cardinality = "ManyToOne"
+	ManyToMany Cardinality = "ManyToMany"
+)
+
+type EntityRelationship struct {
+	ID                string      `json:"id"`
+	Cardinality       Cardinality `json:"cardinality"`
+	SourceEntityID    string      `json:"sourceEntityId"`
+	SourceAttributeID string      `json:"sourceAttributeId"`
```

Comment:
didn't see this in the frontend, what's this needed for?

Reply:
I didn't do any relationships in the frontend client as I was only creating a single entity. It's used for defining relationships between entities (see the confluence docs for more details).

Reply:
yes, the relationship I was aware of, but why do the relationships need attributes? what's that for? and the relationship type in the frontend doesn't have these properties

Reply:
Yeah I want to think through relationships a bit more. I've added a line in the confluence doc that the relationship stuff may be wrong. Previously I was thinking of relationships as being defined by the primary keys, but I was thinking here that they could also be defined on an attribute level. I'm actually leaning back towards doing it based on ids, but I want to spend more time thinking through this again.

As I was typing this one other thought I had, if we were to have a "question bank" then if entities pulled attributes from the question bank we would actually have some implicit relationships. Maybe not useful functionally, but something interesting to see on the analytics side.

Reply:
I think that's gonna bite us in the ass. I would rather add low-level `FOREIGN KEY` indexes in `table` objects rather than trying to find an abstraction for those relationships